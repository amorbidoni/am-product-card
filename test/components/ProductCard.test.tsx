import * as React from 'react';
import  renderer, { act }  from 'react-test-renderer';

import {  ProductCard } from "../../src/components";
import { product1 } from '../data/products';

describe('PoductCard', () => {

    test('debe mostrar el comopnente correctamente', ()=>{

        const wrapper = renderer.create(
            <ProductCard product={ product1 }>
                {
                    ()=>(
                        <h1>Children</h1>
                    )
                }
            </ProductCard>
        )
       expect (wrapper.toJSON()).toMatchSnapshot()
    })

    test('Debe incrementar el contador', ()=>{
        const wrapper = renderer.create(
            <ProductCard product={ product1 }>
                {
                    ({count, increaseBy})=>(
                        <>
                            <h1>Product card</h1>
                            <p>{count}</p>
                            <button onClick={()=> increaseBy(1)}>1</button>
                        </>
                    )
                }
            </ProductCard>
        );
        
        let tree = wrapper.toJSON();
        
        expect(tree).toMatchSnapshot();

        act(()=>{
            (tree as any).children[2].props.onClick(); 
            tree = wrapper.toJSON();
            expect ((tree as any).children[2].children[0]).toBe('1')  
        })



    })
});
