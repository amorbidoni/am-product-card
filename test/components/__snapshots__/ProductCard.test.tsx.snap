// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoductCard Debe incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product card
  </h1>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    1
  </button>
</div>
`;

exports[`PoductCard debe mostrar el comopnente correctamente 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Children
  </h1>
</div>
`;
